### ISSUE TITLE
[BUG] No CLI option to generate a new (additional) receive address (possible in GUI only)

### ISSUE THREAD

**Issue by Crumbles-Pieces** [Chia-Network/chia-blockchain#4869]
I see nowhere in the command line interface to create new wallet receive addresses. It's possible in the GUI though. I'm referring to the infinite receiving addresses that link back to the same wallet.

**Comment by Plopsi** [comment]
`chia wallet get_address`

**Comment by Crumbles-Pieces** [comment]
> `chia wallet get_address`

that just shows me my first wallet address

**Comment by Plopsi** [comment]
whoops.. you are right, this is definitly a bug. change the headline with a starting [BUG] to mark it

**Comment by sargonas** [comment]
Is `chia wallet get_address` not working as intended? My understanding is this should (and does) generate a receiving wallet address for you on the CLI?

**Comment by Crumbles-Pieces** [comment]
> Is `chia wallet get_address` not working as intended? My understanding is this should (and does) generate a receiving wallet address for you on the CLI?

I thought `chia wallet get_address` would create a new receiving address for a wallet. However, when I use that command it just prints a used address from one of my most recent transactions.

**Comment by Plopsi** [comment]
yes, but it generate a new receiving address anytime called, but currently just returns the current one, so always the same initial address

**Comment by sargonas** [comment]
Ahh i see the problem, you are right, just tested it myself, it just repeats the first address each time. There is no CLI to generate a new address.. I just dug through the CLI code in chia/cmd and looking through it all, I'm pretty confident the ability to do so doesn't exist in there anywhere. I'll convert this to a proper bug and flag it.

**Comment by altendky** [comment]
Since I just looked this up separately...

The `False` here:
https://github.com/Chia-Network/chia-blockchain/blob/66c2399ad5cedafed8fa8af7b61d1eda8082859a/chia/cmds/wallet_funcs.py#L111

Could be filled from a CLI option to let the user request a new address via the `new_address` option here:
https://github.com/Chia-Network/chia-blockchain/blob/66c2399ad5cedafed8fa8af7b61d1eda8082859a/chia/rpc/wallet_rpc_client.py#L121

At least, presuming the rest is plumbed up behind the RPC client there.

**Comment by altendky** [comment]
As was done over in https://github.com/Chia-Network/chia-blockchain/pull/4808.

**Comment by wrosko** [comment]
Can we get an update on this?

**Comment by altendky** [comment]
1.3 got a new sub-command for working with keys.

```console
$ chia keys derive wallet-address --help
Usage: chia keys derive wallet-address [OPTIONS]

Options:
  -i, --index INTEGER            Index of the first wallet address to derive.
                                 Index 0 is the first wallet address.

  -n, --count INTEGER            Number of wallet addresses to derive,
                                 starting at index.

  -x, --prefix TEXT              Address prefix (xch for mainnet, txch for
                                 testnet)

  -d, --non-observer-derivation  Derive wallet addresses using non-observer
                                 derivation.  [default: False]

  --show-hd-path                 Show the HD path of the derived wallet
                                 addresses. If non-observer-derivation is
                                 specified, path indices will have an 'n'
                                 suffix.  [default: False]

  -h, --help                     Show this message and exit.
```

**Comment by paninaro** [comment]
With the change that was just merged, you can now use `chia wallet get_address -n` (or `--new-address` for folks who like more typing) to create a new wallet receive address. As @altendky mentioned in https://github.com/Chia-Network/chia-blockchain/issues/4869#issuecomment-1066228981, there's also the `chia keys derive` command available since 1.3.0 that provides a bunch of additional utilities around key derivation. e.g.

`chia keys derive wallet-address -n 10 --show-hd-path`

This can be useful to see which addresses may have historically been used in old transactions. `chia wallet get_address` doesn't tell you what HD path was used for the current derivation.
